addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,374 +1,456 @@
+    appraisal_subordinates_survey_id = fields.Many2one('survey.survey', "collaborate's Appraisal")
+    color = fields.Integer('Color Index')
+    display_name = fields.Char(compute='_set_display_name')
+    mail_template = fields.Many2one('email.template', string="Email Template For Appraisal", default=_set_default_template)
+    email_to = fields.Char('Appraisal Receiver', compute='_set_appraisal_url')
+    appraisal_url = fields.Char('Appraisal URL', compute='_set_appraisal_url')
+    tot_sent_appraisal = fields.Integer('Number of sent appraisal', compute='_get_tot_sent_appraisal', defualt=0, store=True)
+    tot_comp_appraisal = fields.Integer('Number of completed appraisal', defualt=0)
+    user_id = fields.Many2one('res.users', string='Related User', default=lambda self: self.env.uid)
+
+    @api.one
+    @api.depends('employee_id')
+    def _set_display_name(self):
+        self.display_name = self.employee_id.name_related
+
+    @api.onchange('employee_id')
+    def onchange_employee_id(self):
+        if self.employee_id:
+            self.department_id = self.employee_id.department_id
+            self.appraisal_manager = self.employee_id.appraisal_manager
+            self.apprasial_manager_ids = self.employee_id.apprasial_manager_ids
+            self.apprasial_manager_survey_id = self.employee_id.apprasial_manager_survey_id
+            self.appraisal_colleagues = self.employee_id.appraisal_colleagues
+            self.appraisal_colleagues_ids = self.employee_id.appraisal_colleagues_ids
+            self.appraisal_colleagues_survey_id = self.employee_id.appraisal_colleagues_survey_id
+            self.appraisal_self = self.employee_id.appraisal_self
+            self.apprasial_employee = self.employee_id.apprasial_employee or self.employee_id.name
+            self.appraisal_self_survey_id = self.employee_id.appraisal_self_survey_id
+            self.appraisal_subordinates = self.employee_id.appraisal_subordinates
+            self.appraisal_subordinates_ids = self.employee_id.appraisal_subordinates_ids
+            self.appraisal_subordinates_survey_id = self.employee_id.appraisal_subordinates_survey_id
+
+    @api.one
+    @api.constrains('employee_id', 'department_id', 'date_close')
+    def _check_employee_appraisal_duplication(self):
+        """ Avoid duplication"""
+        if self.employee_id and self.department_id and self.date_close:
+            date_closed = datetime.datetime.strptime(self.date_close, DTF)
+            appraisal_ids = self.search([
+                ('employee_id', '=', self.employee_id.id), ('department_id', '=', self.department_id.id),
+                ('date_close', '<=', time.strftime('%Y-' + str(date_closed.month) + '-' + str(date_closed.day))),
+                ('date_close', '>=', time.strftime('%Y-' + str(date_closed.month) + '-01'))])
+            if len(appraisal_ids) > 1:
```

Comment:
It should be len(appraisal_ids) only, here you are allowing to create 1 duplicate appraisal